### PR TITLE
UBC pdportal port from waterlooworks

### PR DESCRIPTION
**Pull request by Ricky-RL**
- functionality ported over to UBC applied sciences co-op board
- removed code related to `satisfaction/rating` pages in pdportal microservice
- currently only supports pulling data from past applications and is not yet integrated with active jobs
- added ability to select pdportal to `main.ts`

### CODE REVIEW

File: src/extractor/index.ts [hunk 1]
```diff
@@ -1,2 +1,5 @@
+
 export { default as waterlooworks } from "./waterlooworks";
 export { default as sfu_myexperience } from "./sfu-myexperience";
+export { default as waterlooworks } from "./pdportal";
```
Comment (jaycsee) [inline src/extractor/index.ts]:
```suggestion
export { default as pdportal } from "./pdportal";
```

File: src/extractor/pdportal/index.ts [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+export * from "./posting";
+export const POSTING_URL =
+  "https://pdportal.apsc.ubc.ca/myAccount/co-op/applications.htm";
+export const DASHBOARD_URL =
+  "https://pdportal.apsc.ubc.ca/myAccount/dashboard.htm";
+
+const POSTING_VIEWPORT = { height: 960, width: 640 };
+
+/**
+ * An extractor for job postings on WaterlooWorks
```
Comment (jaycsee) [inline src/extractor/pdportal/index.ts]: Update name

File: src/extractor/pdportal/posting/data.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+import { parseWorkTerm } from "./workTerm";
+
+export interface ParsablePostingData {
+  id: number;
+  title: string;
+  subtitle: string;
+
+  error: null;
+
+  preScreening: boolean;
```
Comment (jaycsee) [inline src/extractor/pdportal/posting/data.ts]: Is prescreening a thing in pdportal?

File: src/extractor/pdportal/posting/data.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    application: {
+      data: appData,
+      availableInteractions,
+      tags,
+      parsed: {
+        deadline: moment
+          .tz(
+            appString("application deadline", "deadline"),
+            "MMMM DD, YYYY HH:mm a",
+            "America/Toronto"
```
Comment (jaycsee) [inline src/extractor/pdportal/posting/data.ts]: Wrong timzeone 
```suggestion
            "America/Vancouver"
```

File: src/extractor/pdportal/posting/index.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+}
+interface CompanyInformation extends FromData {
+  parsed: {
+    organization: string;
+    division: string;
+  };
+}
+
+/**
+ * An object representing a posting on WaterlooWorks
```
Comment (jaycsee) [inline src/extractor/pdportal/posting/index.ts]: Update name

File: src/extractor/pdportal/posting/status.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  | "AVAILABLE"
+  | "UNKNOWN";
+
+export type InterviewStatus =
+  | "APPLIED"
+  | "NOT-SELECTED"
+  | "SELECTED-FOR-INTERVIEW"
+  | "EMPLOYED"
+  | "NONE"
+  | "UNKNOWN";
```
Comment (jaycsee) [inline src/extractor/pdportal/posting/status.ts:29]: Are these all the same on pdportal?

**Review by jaycsee — CHANGES_REQUESTED** [review]

File: src/extractor/pdportal/posting/data.ts (hunk 3 above)
Comment (Ricky-RL) [reply src/extractor/pdportal/posting/data.ts]: nope lemme get rid of that

File: src/extractor/pdportal/posting/status.ts (hunk 6 above)
Comment (Ricky-RL) [reply src/extractor/pdportal/posting/status.ts:29]: `Internal status`
Most of them look familiar, but currently all the jobs I have access to have an internal status of `expired` so I left them all there to be safe

`Application Status`
yea these are the same

`InterviewStatus`
these are also the same

**Review by jaycsee — APPROVED** [review]
Will reformat and merge

**Pull request merged**